### ISSUE TITLE
Flameshot's gui crashes

### ISSUE THREAD

**Issue by Ieremies** [flameshot-org/flameshot#1721]
**Flameshot version**

```
➜ flameshot -v    
Flameshot v0.9.0
Compiled with Qt 5.15.2
```

**Describe the bug**

When using any flameshot command with gui, it crashes within 2 seconds. Using only `flameshot` it gives me:
```
➜ flameshot         
QSettings::value: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
[1]    18171 segmentation fault (core dumped)  flameshot
```

**To Reproduce**

All of those commands crashes `flameshot`, `flameshot gui`, `flameshot config`, but version and help still works.

**Expected behavior**

A GUI.

**System Information**

Manjaro i3

**Comment by mmahmoudian** [comment]
Please read the [FAQ](https://flameshot.org/guide/faq/) and [Troubleshooting](https://flameshot.org/guide/troubleshooting/) carefully. Considering that you are on i3, my bet is one of the following:
1. Not having systray (no place to show the tray icon)
2. Not having a proper notification daemon (try `pamac install dunst`)

Please close this issue if your problem is solved, if not, tell us what errors you get.

**Comment by Ieremies** [comment]
I do have a systray (currently with clipt and nm-applet).
`notifys-send "test"` works (using dunst).

**Comment by Ieremies** [comment]
I deleted my `flameshot.ini` and ran `flameshot gui`... it works now. Still no clue what happened.

**Comment by mmahmoudian** [comment]
@Ieremies do you happen to have a backup from your config file? It would be interested to know what caused the issue.

**Comment by mmahmoudian** [comment]
@Ieremies Thanks for providing the info. I really cannot wrap my head around why you got segfault with this diff. Let's keep this bug report in mind in case this happened again so that we have some historical evidence of such behavior.

**Comment by kryshac** [comment]
the same thing happened to me
* I deleted the file,
  after that its working
* I put the previous config
  and the same its working

**Comment by kryshac** [comment]
if I set `disabledTrayIcon=true` then I get` segmentation fault` (but you have to close the flameshot before trying)

**Comment by Ieremies** [comment]
I can confirm @kryshac point. Although changing this config in `fameshot config` does not make it crash (and it does hide it form tray).

**Comment by mmahmoudian** [comment]
@kryshac & @Ieremies Similar to #1730

**Comment by chuahou** [comment]
@kryshac 

> the same thing happened to me
> 
>     * I deleted the file,
>       after that its working
> 
>     * I put the previous config
>       and the same its working

> if I set `disabledTrayIcon=true` then I get` segmentation fault` (but you have to close the flameshot before trying)

As you mention in your second message, you have to close flameshot before reproducing the segmentation fault. This is because (as far as I can tell) the segmentation fault occurs when the daemon tries to start, so if the daemon is already running, replacing the previous config does not cause another crash. (It's the bug report @mmahmoudian has kindly linked.)

**Comment by VoxelCubes** [comment]
I'd like to add that I had the same issue. Opening the program with the tray icon disabled would cause a crash, however, once the program was running, you could disable the tray icon once again. That fix would only last until the process was killed, crashing the next time it had to restart. The ini file itself wasn't corrupted, so no issues there.
It appears you have found the culprit, a nullptr would easily explain the segfault.
The official repositories haven't updated the version just yet, but if the issue persists with version 0.10.1+ I'll let you know. Other than that, good job on fixing the issue, I appreciate it.

**Comment by mmahmoudian** [comment]
The v0.10.1 will very soon (today or tomorrow) be released with fixes for this issue. If you are on Arch, installing from `flameshot-git` would already give you the fix.

**Comment by borgmanJeremy** [comment]
Already released 😀

**Comment by ctnguyenvn** [comment]
the same error with new version updated.
```
ctnguyenvn[~]$ flameshot --version
Flameshot v0.10.0
Compiled with Qt 5.15.2
ctnguyenvn[~]$ flameshot
QSettings::value: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
QSettings::value: Empty key passed
QSettings::setValue: Empty key passed
Segmentation fault (core dumped)
```

**Comment by chuahou** [comment]
@ctnguyenvn the error fix is only in v0.10.1 and not v0.10.0, assuming it's the one fixed by #1742.

**Comment by ctnguyenvn** [comment]
> @ctnguyenvn the error fix is only in v0.10.1 and not v0.10.0, assuming it's the one fixed by #1742.

Sorry, I use AUR and don't pay attention to the version (.1).
